website/content/en/docs/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,387 @@
+
+The document considers [memcached operator][memcached-operator] as an example to describe the process of migrating an operator built using Operator SDK to a runnable Kubebuilder project.
+
+## Prerequisites
+- [git][git_tool].
+- [go][go_tool] version 17.03+.
+- [kubectl][kubectl_tool] version v1.11.3+.
+- [kustomize][kustomize_tool] v3.1.0+.
+- Access to a Kubernetes v1.11.3+ cluster.
+
+## Install Kubebuilder
+
+Follow the installation steps provided in the [kubebuilder book][kubebuilder_install] to install kubebuilder.
+
+## Create a new project
+
+Use [`kubebuilder init`][kubebuilder-create-project] command to create a new project:
+
+```sh
+$ mkdir $HOME/projects
+$ cd $HOME/projects
+$ kubebuilder init --domain example.com --license apache2 --owner "The Memcached Operator authors" --repo github.com/example-inc/memcached-operator
+$ cd memcached-operator
+```
+
+Make sure to activate the go module support by running `$ export GO111MODULE=on`.
+
+## Create an API
+
+Create a new API (group/version) as `cache/v1alpha1` and a new kind(CRD) `Memcached` on it.
+
+`kubebuilder create api --group cache --version v1alpha1 --kind Memcached`
+
+Press `y` when asked for creating resource and controller. This will create the files `api/v1alpha1/memcached_types.go` and `controller/memcached_controller.go`
+
+```sh
+Create Resource [y/n]
+y
+Create Controller [y/n]
+y
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_types.go
+controllers/memcached_controller.go
+Running make:
+$ make
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+go fmt ./...
+go vet ./...
+```
+
+### Define Spec and Status
+
+Copy over the API spec and status from the SDK project's [`pkg/apis/cache/v1aplha1/memcached_types.go`][memcached_types] to Kubebuilder's `api/v1aplha1/memcached_types.go`.
+
+For example the `MemcachedSpec` and `MemcachedStatus` which are to be copied are:
+
+```go
+type MemcachedSpec struct {
+	// Specify size, the number of pods
+	Size int32 `json:"size"`
+}
+
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### CRD markers
+
+Include any of the CRD generation or validation markers defined for your API type and fields in `api/v1alpha1/memcached_types.go`
+
+Example:
+```Go
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=memcacheds,scope=Namespaced
+```
+
+### Controller logic
+
+Copy over the `Reconcile()` code and the controller logic from [`pkg/controller/memcached/memcached_controller.go`][memcached_controller] to the new controller's reconcile function in Kubebuilder project at `controllers/memcached_controller.go`.
```

Comment:
So as I mentioned above about rephrasing the migration steps into 1)generic step and 2)example for memcached-operator :
```
Copy over the `Reconcile()` code and the controller logic from your existing controller at 
`pkg/controller/<kind>/<kind>_controller.go` to the new controller's reconcile function 
in Kubebuilder project at `controllers/<kind>_controller.go`.

In our example [`pkg/controller/memcached/memcached_controller.go`][memcached_controller] is copied to `controllers/memcached_controller.go`.
```